pkg/chunked/filesystem_linux.go

```diff
@@ -49,8 +49,31 @@ type fileMetadata struct {
 	skipSetAttrs bool
 }
 
+// splitPath takes a file path as input and returns two components: dir and base.
+// Differently than filepath.Split, this function handles some
+// edge cases.  If the path refers to a file in the root directory, the returned
+// dir is the empty string.
+// The returned base value is never empty, it never contains any slash and the
+// value "..".
+func splitPath(path string) (string, string) {
+	path = filepath.Clean(path)
+	dir, base := filepath.Split(path)
+	if isRootDirectory(dir) {
+		dir = ""
+		if base == ".." {
+			base = "."
+		}
+	} else {
+		dir = removeDotDotPrefix(filepath.Clean(dir))
+	}
+	if base == "" {
+		base = "."
+	}
+	return dir, base
+}
+
 func doHardLink(dirfd, srcFd int, destFile string) error {
-	destDir, destBase := filepath.Split(destFile)
+	destDir, destBase := splitPath(destFile)
 	destDirFd := dirfd
 	if destDir != "" && destDir != "." {
```

Comment:
AFAICS, and per the added documentation, `destDir` can never be `"."` now (adding a few tests to `TestSplitPath` to confirm this would be nice), so in all callers of `splitPath`, this condition can be simplified.

Reply:
thanks, I'll simplify the checks.